.github/workflows/Pitests.yml

```diff
@@ -0,0 +1,635 @@
+name: "Pitest"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  Pitest-main:
+    runs-on: macos-latest
+    steps:
+      - name: Set up JDK 8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+      - name: Setup local maven cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: checkstyle-maven-cache-${{ hashFiles('**/pom.xml') }}
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Pitest-main
+        run: |
+          ./.ci/pitest.sh pitest-main
+      - name: Stage results
+        if: failure()
+        run: |
+          mkdir staging && cp -r target/pit-reports/ staging
```

Comment:
please explain reason of copy report to special folder
if this sub folder in "target" is available in next step we can do archiving from it directly.

Reply:
After a lot of testing, I found that we were not able to directly upload the pit-test folder if the tests fail. I did not found any concrete reason for it. GitHub docs suggest moving artifacts to a new folder and upload them. See: https://docs.github.com/en/actions/guides/building-and-testing-java-with-maven#packaging-workflow-data-as-artifacts